### PR TITLE
Limit the characters to 100 per line in the utility and data modules

### PR DESCRIPTION
**Pull request by vinitamurthi**
## Explanation
Change the utility and data module files to ensure the number of characters per line is 100. Note that only the code lines have been truncated to 100 characters per line. Comments have not changed. Unfortunately, neither ktlint nor Android Studio reformats comment lines. Ktlint doesn't report errors for comment lines either so we would have to check this ourselves

This PR also introduces the --android flag in the ktlint command now, to ensure we follow kotlin android styles.

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ ] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: utility/src/main/java/org/oppia/util/parser/HtmlParser.kt [hunk 1]
```diff
@@ -38,7 +38,11 @@ class HtmlParser private constructor(
     }
 
     if (htmlContent.contains(CUSTOM_IMG_TAG)) {
-      htmlContent = htmlContent.replace(CUSTOM_IMG_TAG, REPLACE_IMG_TAG, /* ignoreCase= */false)
+      htmlContent = htmlContent.replace(
+        CUSTOM_IMG_TAG,
+        REPLACE_IMG_TAG,
+        /* ignoreCase= */false
```
Comment (anandwana001) [inline utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: Should we add space in between `/* ignoreCase= */` and `false`?

Comment (vinitamurthi) [reply utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: Done

File: utility/src/test/java/org/oppia/util/data/DataProvidersTest.kt [hunk 2]
```diff
@@ -70,6 +70,8 @@ class DataProvidersTest {
   @Inject
   lateinit var asyncDataSubscriptionManager: AsyncDataSubscriptionManager
 
+  @Inject lateinit var fakeExceptionLogger: FakeExceptionLogger
```
Comment (anandwana001) [inline utility/src/test/java/org/oppia/util/data/DataProvidersTest.kt]: We are doing `Reformat code`, right? `@Inject` and other stuff in the next line.  

```
@Inject 
lateinit var fakeExceptionLogger: FakeExceptionLogger
```

File: utility/src/test/java/org/oppia/util/data/DataProvidersTest.kt [hunk 3]
```diff
@@ -3226,7 +3227,12 @@ class DataProvidersTest {
 
   // TODO(#89): Move this to a common test application component.
   @Singleton
-  @Component(modules = [TestDispatcherModule::class, TestModule::class, TestLogReportingModule::class])
+  @Component(
+    modules = [
+      TestDispatcherModule::class, TestModule::class,
+      TestLogReportingModule::class
+    ]
```
Comment (anandwana001) [inline utility/src/test/java/org/oppia/util/data/DataProvidersTest.kt:3235]: This seems nice, I will follow this too.

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: utility/src/test/java/org/oppia/util/data/DataProvidersTest.kt (hunk 2 above)
Comment (vinitamurthi) [reply utility/src/test/java/org/oppia/util/data/DataProvidersTest.kt]: Good catch! This was another PR that was merged recently, after merging from develop I didn't reformat and so this got missed. Reformatted now

File: utility/src/main/java/org/oppia/util/parser/HtmlParser.kt [hunk 4]
```diff
@@ -38,7 +38,11 @@ class HtmlParser private constructor(
...
     if (htmlContent.contains(CUSTOM_IMG_TAG)) {
-      htmlContent = htmlContent.replace(CUSTOM_IMG_TAG, REPLACE_IMG_TAG, /* ignoreCase= */false)
+      htmlContent = htmlContent.replace(
+        CUSTOM_IMG_TAG,
+        REPLACE_IMG_TAG,
+        /* ignoreCase= */ false
+      )
       htmlContent = htmlContent.replace(
         CUSTOM_IMG_FILE_PATH_ATTRIBUTE,
         REPLACE_IMG_FILE_PATH_ATTRIBUTE, /* ignoreCase= */false
```
Comment (anandwana001) [inline utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: As discussed with other teammates in my other app module PR, this one comes. 
All should be in a separate line. 

```
htmlContent = htmlContent.replace(
        CUSTOM_IMG_FILE_PATH_ATTRIBUTE,
        REPLACE_IMG_FILE_PATH_ATTRIBUTE,
        /* ignoreCase= */ false
)
```

File: utility/src/main/java/org/oppia/util/parser/HtmlParser.kt [hunk 5]
```diff
@@ -50,10 +54,19 @@ class HtmlParser private constructor(
...
+      htmlContent,
+      HtmlCompat.FROM_HTML_MODE_LEGACY,
+      imageGetter,
+      LiTagHandler()
+    ) as Spannable
 
     val spannableBuilder = SpannableStringBuilder(htmlSpannable)
-    val bulletSpans = spannableBuilder.getSpans(0, spannableBuilder.length, BulletSpan::class.java)
+    val bulletSpans = spannableBuilder.getSpans(
+      /* queryStart= */0,
```
Comment (anandwana001) [inline utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: Space missed `/* queryStart= */ 0`

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: utility/src/main/java/org/oppia/util/parser/HtmlParser.kt (hunk 5 above)
Comment (vinitamurthi) [reply utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: Done

File: utility/src/main/java/org/oppia/util/parser/HtmlParser.kt (hunk 4 above)
Comment (vinitamurthi) [reply utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: Done

File: utility/src/main/java/org/oppia/util/parser/HtmlParser.kt [hunk 6]
```diff
@@ -45,7 +45,8 @@ class HtmlParser private constructor(
       )
       htmlContent = htmlContent.replace(
         CUSTOM_IMG_FILE_PATH_ATTRIBUTE,
-        REPLACE_IMG_FILE_PATH_ATTRIBUTE, /* ignoreCase= */false
+        REPLACE_IMG_FILE_PATH_ATTRIBUTE,
+        /* ignoreCase= */false
```
Comment (anandwana001) [inline utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: space missed `/* ignoreCase= */ false` else everything perfect

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: utility/src/main/java/org/oppia/util/parser/HtmlParser.kt (hunk 6 above)
Comment (vinitamurthi) [reply utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: Oops sorry I missed that! Done now

Comment (anandwana001) [reply utility/src/main/java/org/oppia/util/parser/HtmlParser.kt]: Awesome

**Review by anandwana001 — APPROVED** [review]
Awesome

**Pull request merged**